### DISCUSSION TITLE
Wait until correct key is pressed

### DISCUSSION THREAD

**Discussion by Quinlanptq** [jspsych/jsPsych#1241] [Q&A] [answered]
I am a bit stuck. I want to present a stimulus and record every key press and its RT until the correct key is pressed.

So stim on, start clock, then record every key and RT. The last record will be when the correct key is pressed.

Any advice gratefully received,
Thanks
Philip.

**Comment by jodeleeuw** [comment, marked as answer]
Hi Philip,

You'll need to modify a plugin to do this. `html-keyboard-response` is probably the right starting point.

You can set the `persist` parameter to `true` on the `getKeyboardResponse()` method. This will allow you to gather more than one response.

https://github.com/jspsych/jsPsych/blob/6bcb624f7d97a7d87c69f0f765abff9ff865a33d/plugins/jspsych-html-keyboard-response.js#L123-L128

Every time a response is made it calls this function

https://github.com/jspsych/jsPsych/blob/6bcb624f7d97a7d87c69f0f765abff9ff865a33d/plugins/jspsych-html-keyboard-response.js#L105-L119

You'll want to edit this so that `info.key` and `info.rt` are stored for each key press. I'd suggest creating an array for each and pushing the values onto the array. Then you can add the arrays to the trial data about here:

https://github.com/jspsych/jsPsych/blob/6bcb624f7d97a7d87c69f0f765abff9ff865a33d/plugins/jspsych-html-keyboard-response.js#L90-L95

For consistency with other plugins, you'll want to use `JSON.stringify()` to convert the arrays to their JSON string representation before saving them in the data.

**Comment by Quinlanptq** [comment]
As ever, really helpful Josh thanks Philip.
